File: frontend/src/types/game.ts
export interface TeamBrief {
  id: number
  name: string
  short_name: string
  league: string
}

export interface StarterBrief {
  id: number
  name: string
  era?: number
}

export interface WeatherBrief {
  temperature_c?: number
  weather_main?: string
  wind_speed_ms?: number
  is_raining?: boolean
}

export interface PredictionBrief {
  home_win_prob: number
  predicted_winner: string
  confidence_tier: "high" | "medium" | "low"
  has_explanation: boolean
  predicted_home_score?: number
  predicted_away_score?: number
}

export interface Game {
  id: number
  game_date: string
  game_time?: string
  status: string
  home_team: TeamBrief
  away_team: TeamBrief
  home_starter?: StarterBrief
  away_starter?: StarterBrief
  venue?: string
  prediction?: PredictionBrief
  weather?: WeatherBrief
}

export interface GamesListResponse {
  date: string
  league: string
  games: Game[]
}

File: backend/app/schemas/game.py
from datetime import date, time
from typing import Optional
from pydantic import BaseModel


class TeamBrief(BaseModel):
    id: int
    name: str
    short_name: str
    league: str

    class Config:
        from_attributes = True


class StarterBrief(BaseModel):
    id: int
    name: str
    era: Optional[float] = None

    class Config:
        from_attributes = True


class WeatherBrief(BaseModel):
    temperature_c: Optional[float] = None
    weather_main: Optional[str] = None
    wind_speed_ms: Optional[float] = None
    is_raining: Optional[bool] = None


class PredictionBrief(BaseModel):
    home_win_prob: float
    predicted_winner: str
    confidence_tier: str
    has_explanation: bool
    predicted_home_score: Optional[int] = None
    predicted_away_score: Optional[int] = None


class GameResponse(BaseModel):
    id: int
    game_date: date
    game_time: Optional[str] = None
    status: str
    home_team: TeamBrief
    away_team: TeamBrief
    home_starter: Optional[StarterBrief] = None
    away_starter: Optional[StarterBrief] = None
    venue: Optional[str] = None
    lineup_locked: bool = False
    prediction: Optional[PredictionBrief] = None
    weather: Optional[WeatherBrief] = None


class GamesListResponse(BaseModel):
    date: str
    league: str
    games: list[GameResponse]
